storage/statedb/hasher.go

```diff
@@ -148,6 +187,58 @@ func (h *hasher) hashChildren(original node, db *Database) (node, node) {
 	}
 }
 
+type hashResult struct {
+	index     int
+	collapsed node
+	cached    node
+}
+
+func (h *hasher) hashChildrenFromRoot(original node, db *Database) (node, node) {
+	switch n := original.(type) {
+	case *shortNode:
+		// Hash the short node's child, caching the newly hashed subtree
+		collapsed, cached := n.copy(), n.copy()
+		collapsed.Key = hexToCompact(n.Key)
+		cached.Key = common.CopyBytes(n.Key)
+
+		if _, ok := n.Val.(valueNode); !ok {
+			collapsed.Val, cached.Val = h.hash(n.Val, db, false)
+		}
+		return collapsed, cached
+
+	case *fullNode:
```

Comment:
If the first node is not a full node, you will not benefit from hasher parallelism.

Reply:
Is there any way we can get benefit by parallelization? And that case is not the case we expect.